### PR TITLE
Convert LogsApi to kotlin.

### PR DESCRIPTION
**Pull request by lucaslabari**
## Goal

- Convert LogsApi to kotlin

## Testing

- Relied on existing unit tests. 

NOTE: Please review the last 3 params of `logPushNotification`. I had to make them nullable because if I make them non-nullable the implementation needs to be updated to primitive types on those 3 params and we would be changing the API.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt [hunk 1]
```diff
@@ -121,8 +125,10 @@ void logException(@NonNull Throwable throwable,
      * @param stacktraceElements the stacktrace to remotely log
      * @param severity           the severity level of the log message
      */
-    void logCustomStacktrace(@NonNull StackTraceElement[] stacktraceElements,
-                             @NonNull Severity severity);
+    fun logCustomStacktrace(
+        stacktraceElements: Array<StackTraceElement?>,
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt:129]: Do we actually allow nullable elements in the array?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt [hunk 2]
```diff
@@ -164,12 +174,14 @@ void logCustomStacktrace(@NonNull StackTraceElement[] stacktraceElements,
...
      */
-    void logPushNotification(@Nullable String title,
-                             @Nullable String body,
-                             @Nullable String topic,
-                             @Nullable String id,
-                             @Nullable Integer notificationPriority,
-                             @NonNull Integer messageDeliveredPriority,
-                             @NonNull Boolean isNotification,
-                             @NonNull Boolean hasData);
+    fun logPushNotification(
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt:177]: We should look to deprecate this and replace it with a more normal looking method. The method signature is.... not fantastic.

**Review by bidetofevil — COMMENTED** [review]
I'll defer to Jamie on this, but it looks OK with me as a conversion. We should probably clean these up and deprecate some of the less ideal ones so we can replace them later.

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt (hunk 1 above)
Comment (lucaslabari) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/LogsApi.kt]: Good catch... updated.

**Pull request merged**